# Installing Packages

In [1]:
# Install needed packages
install.packages("psych")
install.packages("quantmod")


The downloaded binary packages are in
	/var/folders/f7/4rwk94x54615k2c8sd0jqjzc0000gn/T//RtmpQkdfAc/downloaded_packages

The downloaded binary packages are in
	/var/folders/f7/4rwk94x54615k2c8sd0jqjzc0000gn/T//RtmpQkdfAc/downloaded_packages


# Importing Libraries

In [2]:
# Import libraries

library(psych)
options(scipen = 999)
library(quantmod)

Loading required package: xts

Loading required package: zoo


Attaching package: ‘zoo’


The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric


Loading required package: TTR

Registered S3 method overwritten by 'quantmod':
  method            from
  as.zoo.data.frame zoo 



# Test 1 - Analysis of SPY and AAPL (Broad stock market)

In [72]:
# Calculate the market beta of a given stock

library(quantmod)

# Define the ticker symbols and time period
stock_symbol <- "AAPL"
index_symbol <- "SPY"
start_date <- "2021-01-01"
end_date <- "2023-01-01"
# end_date <- Sys.Date()
cat("Start date: ", start_date, "\n")
cat("End date: ", end_date, "\n")


# Download the historical stock prices
getSymbols(stock_symbol, from = start_date, to = end_date)
getSymbols(index_symbol, from = start_date, to = end_date)

# Close values

stock_data <- get(stock_symbol)
index_data <- get(index_symbol)

stock_close <- Cl(get(stock_symbol))
index_close <- Cl(get(index_symbol))

# Get returns
stock_returns <- dailyReturn(stock_close)
index_returns <- dailyReturn(index_close)




# summary(stock_data)
# summary(index_data)

# Find Correlation between Stock and Index
print("Correlation")
corr <- cor(stock_close, index_close)
corr

# regression <- lm(stock_close~ index_close )
# summary(regression)

# Volatility and standard deviation of stock

cat("Volatility and SD of Stock:\n")
sd(stock_close)
sd(stock_returns, na.rm=TRUE)


# Volatility and standard deviation of index
cat("Volatility and SD of Index:\n")
sd(index_close)
sd(index_returns, na.rm=TRUE)

# Combine the stock and index returns into a data frame
data <- data.frame(stock_returns, index_returns)

n <- nrow(data)
cat("Sample size: ", n, "\n\n")

# Estimate the linear regression model of the stock returns against the index returns
model <- lm(stock_returns ~ index_returns, data=data)
cat("Summary of Linear Regression Model\n")
summary(model)

# Extract the beta value from the model coefficients
beta <- coef(model)[2]

# Calculate beta using covariance & variance
market_beta <- cov(stock_returns, index_returns) / var(index_returns)

# Print the beta value
cat("The market beta of", stock_symbol, "is", round(beta, 3))
cat("\n")


# calculate critical value for two-tailed test and 95% confidence level
critical_val <- qt(0.975, n-2)

cat("Critical value for 95% confidence level: ", critical_val,"\n\n")

Start date:  2021-01-01 
End date:  2023-01-01 


[1] "AAPL"

[1] "SPY"

[1] "Correlation"


SPY.Close
AAPL.Close 0.5983733

Volatility and SD of Stock:


[1] 15.50367

[1] 0.01941482

Volatility and SD of Index:


[1] 30.14465

[1] 0.01228192

Sample size:  503 

Summary of Linear Regression Model



Call:
lm(formula = stock_returns ~ index_returns, data = data)

Residuals:
      Min        1Q    Median        3Q       Max 
-0.046171 -0.005854 -0.000243  0.005313  0.056873 

Coefficients:
                 Estimate  Std. Error t value            Pr(>|t|)    
(Intercept)   0.000003893 0.000492851   0.008               0.994    
index_returns 1.300223199 0.040165217  32.372 <0.0000000000000002 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.01105 on 501 degrees of freedom
Multiple R-squared:  0.6766,	Adjusted R-squared:  0.6759 
F-statistic:  1048 on 1 and 501 DF,  p-value: < 0.00000000000000022


The market beta of AAPL is 1.3
Critical value for 95% confidence level:  1.96471 



# Test 2 - Analysis of VNQ and O (Real Estate)

In [70]:
# Calculate the market beta of a given stock

library(quantmod)

# Define the ticker symbols and time period
stock_symbol <- "O"
index_symbol <- "IYR"
start_date <- "2021-01-01"
end_date <- "2023-01-01"
# end_date <- Sys.Date()
cat("Start date: ", start_date, "\n")
cat("End date: ", end_date, "\n")


# Download the historical stock prices
getSymbols(stock_symbol, from = start_date, to = end_date)
getSymbols(index_symbol, from = start_date, to = end_date)

# Close values

stock_data <- get(stock_symbol)
index_data <- get(index_symbol)

stock_close <- Cl(get(stock_symbol))
index_close <- Cl(get(index_symbol))

# Get returns
stock_returns <- dailyReturn(stock_close)
index_returns <- dailyReturn(index_close)


# summary(stock_data)
# describe(stock_data)

# summary(index_data)
# describe(index_data)


# Find Correlation between Stock and Index
print("Correlation")
corr <- cor(stock_close, index_close)
corr

# regression <- lm(stock_close~ index_close )
# summary(regression)

# Volatility and standard deviation of stock

cat("Volatility and SD of Stock:\n")
sd(stock_close)
sd(stock_returns, na.rm=TRUE)


# Volatility and standard deviation of index
cat("Volatility and SD of Index:\n")
sd(index_close)
sd(index_returns, na.rm=TRUE)

# Combine the stock and index returns into a data frame
data <- data.frame(stock_returns, index_returns)
# variable to represent sample size n
n <- nrow(data)
cat("Sample size: ", n, "\n\n")

# Estimate the linear regression model of the stock returns against the index returns
model <- lm(stock_returns ~ index_returns, data=data)

# Extract the beta value from the model coefficients
beta <- coef(model)[2]

# Calculate beta using covariance & variance
market_beta <- cov(stock_returns, index_returns) / var(index_returns)

# Print the beta value
cat("The market beta of", stock_symbol, "is", round(beta, 3))
cat("\n")

# Summarize model from LM function
summary(model)

# calculate critical value for two-tailed test and 95% confidence level
critical_val <- qt(0.975, n-2)

cat("Critical value for 95% confidence level: ", critical_val,"\n\n")

Start date:  2021-01-01 
End date:  2023-01-01 


[1] "O"

[1] "IYR"

[1] "Correlation"


IYR.Close
O.Close 0.7731529

Volatility and SD of Stock:


[1] 4.209386

[1] 0.01273373

Volatility and SD of Index:


[1] 9.462274

[1] 0.01272074

Sample size:  503 

The market beta of O is 0.803



Call:
lm(formula = stock_returns ~ index_returns, data = data)

Residuals:
       Min         1Q     Median         3Q        Max 
-0.0304823 -0.0041101  0.0001215  0.0046325  0.0268265 

Coefficients:
               Estimate Std. Error t value            Pr(>|t|)    
(Intercept)   0.0001664  0.0003394    0.49               0.624    
index_returns 0.8028960  0.0267091   30.06 <0.0000000000000002 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.007612 on 501 degrees of freedom
Multiple R-squared:  0.6433,	Adjusted R-squared:  0.6426 
F-statistic: 903.6 on 1 and 501 DF,  p-value: < 0.00000000000000022


Critical value for 95% confidence level:  1.96471 



## Analysis - SPY & AAPL

The actual t value in this case is 32.372
The calculated t value is 1.960393

Since the actual t value is greater than the calculated t value (at 95% confidence), we reject the null hypothesis.

The adjusted r coefficient for this dataset is 0.78

This shows that there is a very strong positive correlation between the performance of the index and the performance of the stock.


## Analysis - O & IYR


The actual t value in this case is 32.372
The calculated t value is 1.960393

Since the actual t value is greater than the calculated t value (at 95% confidence), we reject the null hypothesis.

The adjusted r coefficient for this dataset is 0.78

This shows that there is a very strong positive correlation between the performance of the index and the performance of the stock.
